File: 릿코드/1374-generate-a-string-with-characters-that-have-odd-counts/1374-generate-a-string-with-characters-that-have-odd-counts.js
/**
 * @param {number} n
 * @return {string}
 */
const generateTheString = (n) => {
    const randomAlphabet = () => String.fromCharCode(Math.floor(Math.random() * (122 - 97)) + 97);
    const map = {};
    for(let i=0; i<n; i++) {
        const a = randomAlphabet();
        map[a] = map[a] == null ? 1 : map[a]+1;
    }

    const array = Object.keys(map).filter(key => map[key] % 2 != 0);

    let count = n - array.length;
    if(count != 0) {
        while(true) {
            const a = randomAlphabet();
            if(array.indexOf(a) == -1) {
                if(count % 2 == 0) {
                    for(let i=0; i<count-1; i++) array.push(a);
                    count = 1;
                    continue;
                }
                array.push(a);
                break;
            }
        }
    }
    return array.join("");
}

File: 릿코드/1374-generate-a-string-with-characters-that-have-odd-counts/1374-generate-a-string-with-characters-that-have-odd-counts.java
class Solution {
    public String generateTheString(int n) {
        Map<Character, Integer> map = new HashMap<>();
        for(int i=0; i<n; i++) {
            char a = randomAlphabet();
            map.put(a, map.getOrDefault(a, 0)+1);
        }

        StringBuilder sb = new StringBuilder();
        map.entrySet().stream()
                .filter(entry -> entry.getValue() % 2 != 0)
                .forEach(entry -> sb.append(entry.getKey()));

        int count = n - sb.length();
        if(count != 0) {
            while(true) {
                char a = randomAlphabet();
                if(sb.indexOf(String.valueOf(a)) == -1) {
                   if(count % 2 == 0) {
                       for(int i=0, len=count-1; i<len; i++) sb.append(a);
                       count = 1;
                       continue;
                   }
                   sb.append(a);
                   break;
                }
            }
        }

        return sb.toString();
    }

    private char randomAlphabet() {
        return (char) new Random()
                .ints(97, 122)
                .findFirst()
                .getAsInt();
    }
}